### PR TITLE
Documentation for example module

### PR DESCRIPTION
**Pull request by ltfschoen**
Relates to https://github.com/paritytech/substrate-developer-hub/issues/44

### CODE REVIEW

File: srml/example/README.adoc [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+```
+use <INSERT_MODULE_NAME>;
+
+pub trait Trait: <INSERT_MODULE_NAME>::Trait { }
+```
+
+2. In your module call the `INSERT_MODULE_NAME` module's `example_getter_name()` function to get <EXPLAIN_WHAT_IT_RETURNS>.
+
+```
+const example_value = <INSERT_MODULE_NAME>::Module<T>>::example_getter_name();
```
Comment (bkchr) [inline srml/example/README.adoc]: I don't understand what you want to tell us here.

Comment (ltfschoen) [reply srml/example/README.adoc]: I've pushed a commit to try an make it clearer

Comment (bkchr) [reply srml/example/README.adoc]: So you just want to show standard rust syntax for calling methods?

Comment (ltfschoen) [reply srml/example/README.adoc]: Sorry, i'm not sure what change you're implying

Comment (ltfschoen) [reply srml/example/README.adoc]: I've pushed a commit that removes the steps that a user how to call getter and setter methods

Comment (bkchr) [reply srml/example/README.adoc]: The example you showed was just about "how to call a function".

Comment (ltfschoen) [reply srml/example/README.adoc]: Yes, it was a dismal example. I'm going to try and approach it similar to the example that's evolving in https://github.com/paritytech/substrate/pull/1943

**Comment by ltfschoen** [comment]
@bkchr I've pushed a commit that moves the docs into lib.rs based on review comments, and the docs for the staking module that's been approved. In the docs you can show/hide the template by clicking the "> Details" link as shown in below screenshot.
You can build the docs with `cargo doc --package srml-example --open`
Should I remove the "Implementation Details" and "Extensibility" sections?

Screenshot:

Screen Shot 2019-03-14 at 3 52 51 pm

**Comment by ltfschoen** [comment]
On hold, pending Assets module https://github.com/paritytech/substrate/pull/1945

**Pull request merged**